src/main/java/ay2122s1_cs2103t_w16_2/btbb/ui/MainWindow.java

```diff
@@ -194,6 +203,7 @@ private CommandResult executeCommand(String commandText) throws CommandException
             CommandResult commandResult = logic.execute(commandText);
             logger.info("Result: " + commandResult.getFeedbackToUser());
             resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());
+            updateStatistics();
```

Comment:
Can we use listeners instead, there is no need to update statistics after every command
```suggestion
```

Because order client pie chart statistics will only change when Orders change, we can add a listener in this class
```
    void initializeStatListeners() {
        logic.getFilteredOrderList().addListener((ListChangeListener<Order>) c -> {
            while (c.next()) {
                if (c.wasAdded() || c.wasRemoved() || c.wasUpdated() || c.wasPermutated()) {
                    logger.info("Updating statistics");
                    updateStatistics();
                }
            }
        });
    }
```

Then call it in `UiManager#start()`